### PR TITLE
Add com.abisource.AbiWord

### PR DESCRIPTION
**Pull request by hfiguiere**
Hi,

This is the official submission to add AbiWord stable to Flathub.

Thanks

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build

**Comment by nedrichards** [comment]
bot, build com.abisource.AbiWord

File: com.abisource.AbiWord.json [hunk 1]
```diff
@@ -0,0 +1,218 @@
+{
+    "app-id" : "com.abisource.AbiWord",
+    "branch" : "3.0",
```
Comment (nedrichards) [inline com.abisource.AbiWord.json]: everything on flathub must be on the 'stable' branch.

File: com.abisource.AbiWord.json [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    "runtime" : "org.gnome.Sdk",
+    "runtime-version" : "3.28",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "abiword",
+    "tags" : [
+       "stable"
+    ],
+    "finish-args" : [
+        "--allow=devel",
```
Comment (nedrichards) [inline com.abisource.AbiWord.json]: what do you need this for?

File: com.abisource.AbiWord.json [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+            "name" : "psiconv",
+            "sources" : [
+                {
+                    "type" : "archive",
+                    "url" : "http://www.frodo.looijaard.name/system/files/software/psiconv/psiconv-0.9.9.tar.gz",
+                    "sha256" : "6d51fe79b502a1e277bea275a574ae2db5b1b9d7daef703a8fa3635ae02a8bb0"
+                }
+            ]
+        },
+        {
```
Comment (nedrichards) [inline com.abisource.AbiWord.json]: is this supposed to be empty?

File: com.abisource.AbiWord.json [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+            "name" : "libgda"
+        },
+        {
```
Comment (nedrichards) [inline com.abisource.AbiWord.json]: is this supposed to be empty?

File: com.abisource.AbiWord.json [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+            "name" : "abiword",
+            "config-opts" : [
+                "--enable-plugins=auto"
+            ],
+            "sources" : [
+              {
+                "type" : "git",
+                "url" : "https://github.com/AbiWord/abiword",
+                "branch": "ABI-3-0-0-STABLE"
```
Comment (nedrichards) [inline com.abisource.AbiWord.json]: flathub releases need to be pinned to a specific commit as well, for reproducibility.

**Comment by nedrichards** [comment]
@hfiguiere this is really cool! delighted to see abiword here.

File: com.abisource.AbiWord.json [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+                "/lib/libboost_*.a"
+            ],
+            "sources" : [
+                {
+                    "type" : "archive",
+                    "url" : "https://downloads.sourceforge.net/project/boost/boost/1.60.0/boost_1_60_0.tar.bz2",
+                    "sha256" : "686affff989ac2488f79a97b9479efb9f2abae035b5ed4d8226de6857933fd3b"
+                },
+                {
+                    "type" : "file",
```
Comment (barthalion) [inline com.abisource.AbiWord.json]: Use the `simple` buildsystem instead; also build only specific libraries Abiword uses.

File: com.abisource.AbiWord.json [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+                    "path" : "boost-configure",
+                    "dest-filename" : "configure"
+                }
+            ]
+        },
+        "fribidi-0.19.7.json",
```
Comment (barthalion) [inline com.abisource.AbiWord.json]: Any reason to store it separately?

File: com.abisource.AbiWord.json [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+        "--filesystem=host",
+        "--share=network",
+        "--talk-name=org.gtk.vfs.*"
+    ],
+    "build-options" : {
+        "env" : {
+            "V" : "1"
+        }
+    },
+    "cleanup" : [
```
Comment (barthalion) [inline com.abisource.AbiWord.json]: You should also cleanup `/include`, but there's surely more unneeded files.

File: com.abisource.AbiWord.json [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+    "runtime" : "org.gnome.Sdk",
+    "runtime-version" : "3.28",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "abiword",
+    "tags" : [
+       "stable"
+    ],
+    "finish-args" : [
+        "--allow=devel",
+        "--talk-name=org.freedesktop.Flatpak",
```
Comment (barthalion) [inline com.abisource.AbiWord.json]: What do you need it for?

File: com.abisource.AbiWord.json [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=host",
```
Comment (barthalion) [inline com.abisource.AbiWord.json]: `xdg-document` would be much better, or making use of xdg-desktop-portal, especially if it's gtk3.

File: com.abisource.AbiWord.json (hunk 6 above)
Comment (barthalion) [reply com.abisource.AbiWord.json]: Some example: https://github.com/flathub/org.qbittorrent.qBittorrent/blob/4f8e78558fd83ea4f684848e69e2f6e00f02808a/org.qbittorrent.qBittorrent.json#L28

Except now there is $FLATPAK_BUILDER_N_JOBS env var to specify number of jobs.

**Comment by barthalion** [comment]
Failed to build: `aclocal: error: configure.ac:211: file 'plugin-list.m4' does not exist`

File: com.abisource.AbiWord.json (hunk 1 above)
Comment (hfiguiere) [reply com.abisource.AbiWord.json]: done

File: com.abisource.AbiWord.json (hunk 2 above)
Comment (hfiguiere) [reply com.abisource.AbiWord.json]: I think this is leftover from Gnome Builder. I copied the .json over. Removed for the next push.

File: com.abisource.AbiWord.json (hunk 3 above)
Comment (hfiguiere) [reply com.abisource.AbiWord.json]: I may I forgotten them. I'll deal with these optional dependencies later.

File: com.abisource.AbiWord.json (hunk 4 above)
Comment (hfiguiere) [reply com.abisource.AbiWord.json]: same

File: com.abisource.AbiWord.json (hunk 5 above)
Comment (hfiguiere) [reply com.abisource.AbiWord.json]: Added the commit.

File: com.abisource.AbiWord.json (hunk 6 above)
Comment (hfiguiere) [reply com.abisource.AbiWord.json]: I changed this now for the next push.

File: com.abisource.AbiWord.json (hunk 7 above)
Comment (hfiguiere) [reply com.abisource.AbiWord.json]: convenience. I had it in a separate json.

File: com.abisource.AbiWord.json (hunk 9 above)
Comment (hfiguiere) [reply com.abisource.AbiWord.json]: leftovers. removed.

**Comment by hfiguiere** [comment]
I fixed the build issue. Was my fault, I was assuming the m4 did exist.

This version should address all comments I think.

File: com.abisource.AbiWord.json (hunk 10 above)
Comment (hfiguiere) [reply com.abisource.AbiWord.json]: I need to actually check AbiWord works with portals.

Comment (nedrichards) [reply com.abisource.AbiWord.json]: The major thing to look at as a GTK app is if you use FileChooserNative (you should)

**Comment by nedrichards** [comment]
bot, build com.abisource.AbiWord

File: boost-configure [hunk 11]
```diff
@@ -0,0 +1,11 @@
+
```
Comment (barthalion) [inline boost-configure]: Can be removed after merge.

Comment (hfiguiere) [reply boost-configure]: I already removed it in my local tree.

File: com.abisource.AbiWord.json (hunk 10 above)
Comment (hfiguiere) [reply com.abisource.AbiWord.json]: AbiWord doesn't as the requiremnt for Gtk+ is 3.12 in the stable branch. I'm ok bumpimg to 3.20 for the unstable branch though.

File: boost-configure (hunk 11 above)
Comment (hfiguiere) [reply boost-configure]: Pushed the commit now.

File: com.abisource.AbiWord.json (hunk 10 above)
Comment (hfiguiere) [reply com.abisource.AbiWord.json]: I use --filesystem=home for now

**Comment by barthalion** [comment]
Build failed because of missing appdata: https://flathub.org/builds/#/builders/2/builds/3273

**Comment by hfiguiere** [comment]
Argh. I thought I had it and realize that it is only in trunk. I'll push it to the stable branch as well and update. Sorry.

Thanks !

**Comment by hfiguiere** [comment]
AppData should be in now.

**Comment by barthalion** [comment]
bot, build com.abisource.AbiWord

**Comment by barthalion** [comment]
Appdata is being installed under wrong name – it needs to match the app id.

**Comment by nedrichards** [comment]
bot, build com.abisource.AbiWord

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.abisource.AbiWord

**Pull request closed without merging**